server/base/src/main/java/org/apache/accumulo/server/compaction/FileCompactor.java

```diff
@@ -280,7 +283,13 @@ public CompactionStats call() throws IOException, CompactionCanceledException {
           }
         }
       } catch (IOException | RuntimeException e) {
-        log.warn("{}", e.getMessage(), e);
+        // If compaction was cancelled then this may happen due to an
+        // InterruptedException etc so suppress logging
+        if (!env.isCompactionEnabled()) {
+          log.debug("{}", e.getMessage(), e);
+        } else {
+          log.warn("{}", e.getMessage(), e);
+        }
```

Comment:
I'm very confused about the negation and order of these. Getting rid of the negation, it becomes:

```suggestion
        // If compaction was cancelled then this may happen due to an
        // InterruptedException etc so suppress logging
        if (env.isCompactionEnabled()) {
          log.warn("{}", e.getMessage(), e);
        } else {
          log.debug("{}", e.getMessage(), e);
        }
```

But in addition to this, the comment is a bit confusing, because it talks about `if compaction was cancelled`, but the code says `if compaction is enabled`. It would be good to improve the comment so it connects the dots between the condition the code is checking and the phrasing of the comment. It's not clear to me from reading this the connection between compaction being enabled and a cancellation.

Reply:
Suggestion:

```
The CompactionEnv may disable the compaction for a number of reasons (e.g. user compaction cancelled, tablet split, table deletion, etc.). When this happens it's possible for an exception to be thrown when a file fails to close or some other cleanup-type issue. These error messages can be supressed.
```

Reply:
Ignore my last, I was making the comment at the same time you pushed a change. Your new comment is essentially the same.

Reply:
sounds good, your is a bit more concise but I think either is fine and as you said basically says the same thing.